packages/mui-joy/src/styles/components.d.ts

```diff
@@ -500,6 +505,10 @@ export interface Components<Theme = unknown> {
     defaultProps?: Partial<SnackbarProps>;
     styleOverrides?: StyleOverrides<SnackbarSlot, SnackbarOwnerState, Theme>;
   };
+  JoyToggleButtonGroup?: {
+    defaultProps?: Partial<ToggleButtonGroupProps<any>>;
+    styleOverrides?: StyleOverrides<ToggleButtonGroupSlot, ToggleButtonGroupOwnerState<any>, Theme>;
```

Comment:
I wonder if we should provide default generic values where the types are defined:
```diff
--- a/packages/mui-joy/src/ToggleButtonGroup/ToggleButtonGroupProps.ts
+++ b/packages/mui-joy/src/ToggleButtonGroup/ToggleButtonGroupProps.ts
@@ -111,10 +111,10 @@ export interface ToggleButtonGroupTypeMap<
 }

 export type ToggleButtonGroupProps<
-  TValue extends SupportedValue,
+  TValue extends SupportedValue = SupportedValue,
   D extends React.ElementType = ToggleButtonGroupTypeMap<TValue>['defaultComponent'],
   P = { component?: React.ElementType },
 > = OverrideProps<ToggleButtonGroupTypeMap<TValue, P, D>, D>;

-export interface ToggleButtonGroupOwnerState<TValue extends SupportedValue>
+export interface ToggleButtonGroupOwnerState<TValue extends SupportedValue = SupportedValue>
   extends ApplyColorInversion<ToggleButtonGroupProps<TValue>> {}
```


That way we won't need to provide the generic parameter types:
```suggestion
    defaultProps?: Partial<ToggleButtonGroupProps>;
    styleOverrides?: StyleOverrides<ToggleButtonGroupSlot, ToggleButtonGroupOwnerState, Theme>;
```